systems-analysis/dfcache-stream-support.md

```diff
@@ -0,0 +1,326 @@
+
+## Motivation
+
+- **Performance**: Streaming gRPC interfaces reduce I/O overhead and enable efficient large file transfers compared to non-streaming methods.
+- **Flexibility**: Streaming support enhances interactivity with progress tracking and task control.
+- **Compatibility**: UDS and Kubernetes optimizations address file path and permission challenges in containerized environments.
+- **User Experience**: Improved progress indicators and error handling enhance usability.
+
+## Design Goals
+
+1. Add streaming gRPC support (`WritePersistentCacheTask` and `ReadPersistentCacheTask`) while maintaining compatibility with existing `dfcache` functionality.
+2. Preserve the simplicity and maintainability of the `dfcache` client architecture.
+3. Optimize UDS connections for performance and reliability in Kubernetes.
+4. Enhance file transfer efficiency with piece-based streaming and robust error recovery.
+
+## Architecture
+
+### Current Architecture
+
+```
+dfcache -> {
+    main.rs: CLI entry with clap parsing
+    import.rs: ImportCommand with UploadPersistentCacheTaskRequest
+    export.rs: ExportCommand with DownloadPersistentCacheTaskRequest
+    stat.rs: StatCommand
+}
+DfdaemonDownloadClient -> gRPC (non-streaming)
+```
+
+### Proposed Architecture
+
+```
+dfcache -> {
+    main.rs: CLI entry with clap parsing
+    import.rs: ImportCommand with WritePersistentCacheTask (streaming)
+    export.rs: ExportCommand with ReadPersistentCacheTask (streaming)
+    stat.rs: StatCommand
+}
+DfdaemonDownloadClient -> gRPC (streaming and non-streaming)
+```
+
+### Module Structure
+
+#### Modified Modules
+
+- `dragonfly-client/src/grpc/dfdaemon_download.rs`: Add streaming gRPC methods (`write_persistent_cache_task`, `read_persistent_cache_task`).
+- `dragonfly-client/src/cli/import.rs`: Refactor `ImportCommand` for streaming uploads.
+- `dragonfly-client/src/cli/export.rs`: Refactor `ExportCommand` for streaming downloads.
+- `dragonfly-client/src/storage/content.rs`: Enhance file read/write with streaming optimizations.
+- Configuration file: Add streaming-specific options.
+
+## Technical Implementation
+
+### 1. Core Interfaces
+
+#### Streaming gRPC Interfaces
+
+```protobuf
+// Write persistent cache task to P2P network
+rpc WritePersistentCacheTask(stream WritePersistentCacheTaskRequest) returns (WritePersistentCacheTaskResponse);
+
+message WritePersistentCacheTaskRequest {
+  oneof response {
+    WritePersistentCacheTaskStartedRequest write_persistent_cache_task_started_request = 1;
+    WritePersistentCacheTaskFinishedRequest write_persistent_cache_task_finished_request = 2;
+    WriteChunkRequest write_chunk_request = 3;
```

Comment:
What is the meaning of the chunk here, is it the chunk of the persistent cache task, if so, please align the name completely.